src/platform/silabs/wifi/SiWx/WifiInterface.cpp

```diff
@@ -462,6 +462,10 @@ sl_status_t JoinWifiNetwork(void)
 
     if (status == SL_STATUS_OK)
     {
+#if CHIP_CONFIG_ENABLE_ICD_SERVER
+        // Remove High performance request that might have been added during the connect/retry process
+        chip::DeviceLayer::Silabs::WifiSleepManager::GetInstance().RemoveHighPerformanceRequest();
+#endif // CHIP_CONFIG_ENABLE_ICD_SERVER
```

Comment:
I'm not sure i understand why we remove a high performance request here.
We should only remove when we have connected to the wifi network.

I don't think this should here.

Reply:
We enabled the high-performance request before the join call and are removing it here if the connection is successful.